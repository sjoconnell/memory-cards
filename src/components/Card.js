import React from "react";
import "./styles/Card.scss";

const Card = ({ value, id, flipped, matched, disabled, handleCardClick }) => {
  return (
    <div
      className={`card ${flipped || matched ? "flip" : ""} ${
        matched ? "matched" : ""
      }`}
      onClick={() =>
        disabled || matched || flipped ? null : handleCardClick(id)
      }
    >
      <div className="card-front">
        <span role="img" aria-label="front-card-icon">
          {value}
        </span>
      </div>
      <div className="card-back">
        <span role="img" aria-label="back-card-icon">
          ❓
        </span>
      </div>
    </div>
  );
};

export default Card;
