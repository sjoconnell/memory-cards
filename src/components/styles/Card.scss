.card {
  width: 90px;
  height: 90px;
  justify-content: center;
  align-items: center;
  display: flex;
  position: relative;
  transform: scale(1);
  transform-style: preserve-3d;
  transition: transform 0.5s;
  margin: 5px;
  font-size: 40px;

  &:active {
    transform: scale(0.97);
    transition: transform 0.2s;
  }

  &.matched {
    background-color: blue;
    transition: transform 0.5s, background-color 3s ease-in-out 0.5s;
    div {
      box-shadow: none;
      transition: box-shadow 2s ease-in-out 0.5s;
    }
    &:hover {
      cursor: default;
    }
  }

  &:hover {
    cursor: pointer;
  }

  &.flip {
    transform: rotateY(180deg);
  }

  .card-front,
  .card-back {
    position: absolute;
    backface-visibility: hidden;
    height: 100%;
    width: 100%;
    background-color: lightblue;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 5px;
    box-shadow: 5px 5px 5px black;
  }

  .card-back {
    transform: rotateY(0deg);
  }

  .card-front {
    transform: rotateY(180deg);
  }
}
