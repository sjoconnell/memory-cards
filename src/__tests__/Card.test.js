import React from "react";
import { create } from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Card from "../components/Card";

configure({ adapter: new Adapter() });

test("Card snapshot", () => {
  const c = create(<Card />);
  expect(c.toJSON()).toMatchSnapshot();
});

test("Should call handleCardClick function when clicking Card", () => {
  const clickCard = jest.fn();
  const wrapper = shallow(<Card handleCardClick={clickCard} />);
  const div = wrapper.find(".card");
  div.simulate("click");
  expect(clickCard).toHaveBeenCalledTimes(1);
});

test("Should not call handleCardClick when disabled is true", () => {
  const clickCard = jest.fn();
  const wrapper = shallow(<Card handleCardClick={clickCard} disabled={true} />);
  const div = wrapper.find(".card");
  div.simulate("click");
  expect(clickCard).toHaveBeenCalledTimes(0);
});

test("Should not call handleCardClick when flipped is true", () => {
  const clickCard = jest.fn();
  const wrapper = shallow(<Card handleCardClick={clickCard} flipped={true} />);
  const div = wrapper.find(".card");
  div.simulate("click");
  expect(clickCard).toHaveBeenCalledTimes(0);
});

test("Should not call handleCardClick when matched is true", () => {
  const clickCard = jest.fn();
  const wrapper = shallow(<Card handleCardClick={clickCard} matched={true} />);
  const div = wrapper.find(".card");
  div.simulate("click");
  expect(clickCard).toHaveBeenCalledTimes(0);
});
