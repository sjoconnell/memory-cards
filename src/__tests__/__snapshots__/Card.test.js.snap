// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot 1`] = `
<div
  className="card  "
  onClick={[Function]}
>
  <div
    className="card-front"
  >
    <span
      aria-label="front-card-icon"
      role="img"
    />
  </div>
  <div
    className="card-back"
  >
    <span
      aria-label="back-card-icon"
      role="img"
    >
      ‚ùì
    </span>
  </div>
</div>
`;
